src/dataregistry/registrar/base_table_class.py

```diff
@@ -99,9 +106,39 @@ def modify(self, entry_id, modify_fields):
             and value is the desired new value for the entry
         """
 
-        raise NotImplementedError
+        assert (
+            type(modify_fields) == dict
+        ), f"modify_fields is expected as a dict, {'column': new_values}"
+
+        # First make sure the given entry is in the registry
+        my_table = self._get_table_metadata(self.which_table)
+        previous_entry = self.find_entry(entry_id)
+
+        # Check entry exists
+        if previous_entry is None:
+            raise ValueError(f"{self.which_table} {entry_id} does not exist")
+
+        # Loop over each column to be modified
+        for key, v in modify_fields.items():
+            # Make sure the column is in the schema
+            if key not in self.schema_yaml[self.which_table].keys():
+                raise ValueError(f"The column {key} doesnt not exist in the schema")
+
+            # Make sure the column is modifiable
+            if not self.schema_yaml[self.which_table][key]["modifiable"]:
+                raise ValueError(f"The column {key} is not modifiable")
+
+        # Update the entries
+        with self._engine.connect() as conn:
+            update_stmt = (
+                update(my_table)
+                .where(getattr(my_table.c, self.entry_id) == entry_id)
+                .values(modify_fields)
+            )
+            conn.execute(update_stmt)
+            conn.commit()
 
-    def find_entry(self, entry_id):
+    def find_entry(self, entry_id, only_valid=False, only_on_disk=False):
```

Comment:
Not sure these are the right defaults; something to discuss.

Reply:
Yes, this was largely me being lazy to reduce the checks at the other end.

The spirit of this function should just be to find an entry regardless of other factors.

I've removed the `only_valid` and `only_on_disk` options. I've added a `raise_if_not_found=False` optional tag to raise an exception if the dataset is not found.

I've moved the "is it already deleted" check back to the `delete` function.